components/service/glean/src/main/java/mozilla/components/service/glean/Dispatchers.kt

```diff
@@ -64,6 +100,76 @@ internal object Dispatchers {
         fun setTestingMode(enabled: Boolean) {
             testingMode = enabled
         }
+
+        /**
+         * Enable testing mode, which makes all of the Glean public API synchronous.
+         *
+         * @param enabled whether or not to enable the testing mode
+         */
+        @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+        fun enableQueueing(enabled: Boolean) {
+            queueInitialTasks = enabled
+        }
+
+        /**
+         * Stops queueing tasks and processes any tasks in the queue. Since [queueInitialTasks] is
+         * set to false prior to processing the queue, newly launched tasks should be executed
+         * on the couroutine scope rather than added to the queue.
+         */
+        fun flushQueuedInitialTasks() {
+            queueSemaphore.acquire(2)
```

Comment:
What if we have 3-level nesting? We'd deadlock again. I think a semaphore might not be a good solution here. We should really go back to the drawing board and think about this thoroughly. Moreover, what's up with the solution of sending a "flipping task"?

Reply:
Semaphore idea was just to illustrate the problem more than provide a solution.  I've backed out those changes per our discussion.